### PR TITLE
Move subscription engines to a sub-folder

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
This pr reorganises a bit the code for subscriptions engines

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-4066.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 35 | 49 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 84 | 110 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 55 | 96 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3856 | 3867 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3336 | 3416 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 59 | 59 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 79 | 60 | 2174 |
|  aggregations.DeeplyNestedCount | 11062443 | 11062443 | 5444 | 5243 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 184 | 212 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 152 | 88 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 171 | 239 | 3003 |
|  connections.Connection | 14082 | 14082 | 151 | 105 | 2174 |
|  connections.NestedConnection | 41118 | 41118 | 158 | 122 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 75 | 113 | 1040 |
|  connections.ConnectionWithSortAndCypher | 16325 | 16325 | 112 | 164 | 2174 |
|  create.SimpleMutation | 7 | 7 | 69 | 104 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 49 | 32 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 257 | 253 | 1040 |
|  delete.NestedDeleteInUpdate | 19542 | 19542 | 278 | 209 | 1183 |
|  2871.NestedRelationshipFilter | 23113 | 23184 | 77 | 88 | 4428 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 45 | 64 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 88 | 120 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 43 | 62 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 65 | 89 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 33 | 30 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 48 | 79 | 2174 |
|  query.QueryWhere | 9708 | 9725 | 45 | 48 | 2164 |
|  query.SimpleQueryWithNestedWhere | 9886 | 9903 | 67 | 67 | 2164 |
|  query.Nested | 10096041 | 10096041 | 11053 | 10291 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10725 | 9757 | 2004000 |
|  query.OrFilterOnRelationships | 43044 | 42146 | 197 | 263 | 1856 |
|  query.OrFilterOnRelationshipsAndNested | 36783 | 35383 | 282 | 349 | 1856 |
|  query.QueryWithNestedIn | 13680 | 13299 | 65 | 84 | 1310 |
|  query.NestedConnectionWhere | 9834 | 9834 | 62 | 71 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9976 | 10044 | 109 | 99 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19128 | 19260 | 160 | 174 | 1596 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 183 | 273 | 1134 |
|  query.Fulltext | 80 | 80 | 41 | 42 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 51 | 58 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 75 | 104 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14437 | 14521 | 103 | 128 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 66 | 69 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 103 | 132 | 4516 |
|  sorting.SortWithTopLevelCypher | 3396 | 3396 | 86 | 85 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 87 | 84 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 67 | 129 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 348 | 534 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 406 | 364 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 113 | 201 | 2002 |

</details>
        
Old Schema Generation: 52.051s
Schema Generation: 48.487s
Old Subgraph Schema Generation: 58.760s
Subgraph Schema Generation: 58.810s

File: packages/graphql/src/types/index.ts [hunk 1]
```diff
@@ -330,7 +331,7 @@ export type NodeSubscriptionsEvent =
               old: undefined;
               new: Record<string, any>;
           };
-          id: number;
+          id: number; // | string
```
Comment (a-alle) [inline packages/graphql/src/types/index.ts:334]: what do these comments mean?

Comment (angrykoala) [reply packages/graphql/src/types/index.ts:334]: These types are not correct, as these types may be strings. I'll have to update these for CDC, but I haven't fixed it yet. I can remove these comments for now

**Review by a-alle — APPROVED** [review]

**Pull request merged**